tracer/src/Datadog.Trace.BenchmarkDotNet/DatadogDiagnoser.cs

```diff
@@ -20,15 +19,12 @@ namespace Datadog.Trace.BenchmarkDotNet;
 /// <summary>
 /// Datadog BenchmarkDotNet Diagnoser
 /// </summary>
-public class DatadogDiagnoser : IDiagnoser
+internal class DatadogDiagnoser : IDiagnoser
```

Comment:
Nobody look, just making breaking changes here 😛 

In all seriousness, I really don't think you can do this in a minor version bump. It _could_ break people, and it's fundamentally not semver. Best you can do is mark it `[obsolete]`, though even that is technically a breaking change. Do you _have_ to change the public API here?

Reply:
I've reverted this visibility change in https://github.com/DataDog/dd-trace-dotnet/pull/4628/commits/4dce0d4c8660c6207771aee22f7cb54992ed04cf I realize that I could be helpful to use the diagnoser in other contexts outside the Extension methods.